### DISCUSSION TITLE
How to get CSV output for Get-VM and Get-VMNetworkAdapter properties using single PowerShell script?

### DISCUSSION THREAD

**Discussion by Srini1811** [PowerShell/PowerShell#15038] [Q&A]
`PS C:\User> Get-VM -Name * | Select-Object -Property Name,State,CPUUsage | Export-Csv -Path E:\MYfolder/report.csv -NoTypeInformation              #giving csv expected results alone
PS C:\User> Get-VMNetworkAdapter -Name * | Select-Object -Property MacAddress,IPAddresses | Export-Csv -Path E:\MYfolder/report.csv -NoTypeInformation         #giving csv expected results alone`

I am newly started my coding and stuck with building a PowerShell above script. How to write a single script to execute Get-VM and Get-VMNetworkAdapter with designated properties. Trying to Get VM information's from Hyper-V host or windows server. Output: should be a single csv file.

**Comment by powercode** [comment]
I'm not sure I understand what kind of output you want. Can you give an example of desired output?

**Reply by Srini1811** [reply]
Hi @powercode  Thank you for replying..
For Example, Get-VM propery(Name|State|CPUUsage),Get-VMNetworkAdapter property(MacAddress|IPAddress) should be write in single script Name: script.ps1 -->
PS> .\script.ps1 -HostName1  or VMname
Output: - Export/List all property results in Single CSV file/ csv should contains below property results | please refer the attachment.

Vmname | State | CPUUsage | MacAddress | IPAddress | HostName | etc
-- | -- | -- | -- | -- | -- | --
VM1 | Running | 2048123 | 33211233213 | 10.130.10.11 |  
VM2 | Running | 2048123 | 33211233213 | 10.130.10.12 |  
VM3 | Running | 2048123 | 33211233213 | 10.130.10.13 |  
VM4 | Running | 2048123 | 33211233213 | 10.130.10.14 |  
  | Expected output like this csv |   |   |   |   |  

![Output-ps1](https://user-images.githubusercontent.com/22885121/111400450-61233300-86ed-11eb-87e3-295667eecff2.JPG)

**Comment by powercode** [comment]
Maybe something like this? Beware - has not been run!

```powershell
function Export-VMInfoCsv {
    param(
        [Parameter(Mandatory)]
        [string] $OutputPath
    )
    $networkAdapters = Get-VMNetworkAdapter -Name * | Group-Object -Property Name -AsHashTable
    Get-VM -Name *
    | ForEach-Object { 
        $na = $networkAdapters[$_.Name]
        Add-Member -InputObject $_ -PassThru -NotePropertyMembers @{
            MacAddress  = $na.MacAddress
            IPAddresses = $na.IPAddresses
        }
    }
    | Select-Object -Property Name, State, CPUUsage, MacAddress, IPAddresses  
    | Export-Csv -Path $OutputPath -NoTypeInformation 
}
```

**Reply by Srini1811** [reply]
Thank you @powercode Code helped to go next step of my code and it's working for me.
